contracts/ERC721RExample.sol

```diff
@@ -75,26 +77,37 @@ contract ERC721RExample is ERC721A, Ownable {
         }
     }
 
-    function refund(uint256[] calldata tokenIds) external {
-        require(isRefundGuaranteeActive(), "Refund expired");
+    function refund(uint256 tokenId) external {
+        require(block.number < refundDeadlineOf(tokenId), "Refund expired");
+        require(msg.sender == ownerOf(tokenId), "Not token owner");
 
-        for (uint256 i = 0; i < tokenIds.length; i++) {
-            uint256 tokenId = tokenIds[i];
-            require(msg.sender == ownerOf(tokenId), "Not token owner");
-            require(!hasRefunded[tokenId], "Already refunded");
-            require(!isOwnerMint[tokenId], "Freely minted NFTs cannot be refunded");
-            hasRefunded[tokenId] = true;
-            transferFrom(msg.sender, refundAddress, tokenId);
-        }
+        hasRefunded[tokenId] = true;
+        _transfer(msg.sender, refundAddress, tokenId);
 
-        uint256 refundAmount = tokenIds.length * mintPrice;
+        uint256 refundAmount = refundOf(tokenId);
```

Comment:
the idea here is that we no longer need the for loop because we have multicall if people want to refund multiple nfts in one go?

Reply:
Yes.